test/Microsoft.Azure.WebJobs.Host.FunctionalTests/Queues/QueueListenerTests.cs

```diff
@@ -20,77 +20,74 @@ namespace Microsoft.Azure.WebJobs.Host.FunctionalTests.Queues
 {
     public class QueueListenerTests : IClassFixture<QueueListenerTests.TestFixture>
     {
+        private TraceWriter _trace;
+        private CloudQueue _queue;
+        private CloudQueue _poisonQueue;
+
+        private JobHostQueuesConfiguration _queuesConfig;
+
+        private StorageQueue _storageQueue;
+        private StorageQueue _storagePoisonQueue;
+        private Mock<ITriggerExecutor<IStorageQueueMessage>> _mockTriggerExecutor = new Mock<ITriggerExecutor<IStorageQueueMessage>>(MockBehavior.Strict);
+
         public QueueListenerTests(TestFixture fixture)
         {
-            Fixture = fixture;
+            _trace = new TestTraceWriter(TraceLevel.Verbose);
+            _queue = fixture.CreateNewQueue();
+            _poisonQueue = fixture.CreateNewQueue();
+            _queuesConfig = new JobHostQueuesConfiguration { MaxDequeueCount = 2 };
```

Comment:
Have you taken a look at QueueProcesorTests? Those are isolated tests for QueueProcessor - I think it makes more sense for your new test to be there. E.g. it should be easy then for you to have a test that calls directly into BeginProcessingMessageAsync with a message whose dequeue count is over limit.